filtering-tags-problem/main.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import os
+from core import azure_wrapper, scheduler
+
+
+def parse_operation():
+    if cli_options["list_blobs"]:
+        return "list_blobs"
+    if cli_options["delete_blobs"]:
+        return "delete_blobs"
+    elif cli_options["download_blobs"]:
+        return "download_blobs"
+    elif cli_options["add_tag"]:
+        return "add_tag"
+    elif cli_options["update_tag"]:
+        return "update_tag"
+    elif cli_options["delete_tag"]:
+        return "delete_tag"
+    else:
+        raise Exception('Invalid option chosen')
+
+
+def main():
+    credentials = {}
+    if cli_options["connection_string"]:
+        credentials["connection_string"] = os.getenv("AZURE_STORAGE_CONNECTION_STRING")
+        # credentials["connection_string"] = cli_options["connection_string"]
+    elif cli_options["sas_url"]:
+        credentials["sas_url"] = cli_options["sas_url"]
+
+    options = {
+        "container_name": cli_options["container"],
+        "name_starts_with": cli_options["name_starts_with"],
+        "tag_key": cli_options["tag_key"],
+        "tag_value": cli_options["tag_value"],
+        "concurrency": cli_options["concurrency_factor"],
```

Comment:
It doesn't appear that we're utilizing the concurrency parameter in the operations. I assume this parameter is for the download operation? if so it doesn't look like its being passed from the options bag.